pkg/scheduler/scheduler.go

```diff
@@ -373,6 +373,13 @@ func buildQueueingHintMap(es []framework.EnqueueExtensions) internalqueue.Queuei
 	for _, e := range es {
 		events := e.EventsToRegister()
 
+		// This will happen when plugin registers with empty events, it's usually the case a pod
+		// will become reschedulable only for self-update, e.g. schedulingGates plugin, the pod
+		// will enter into the activeQ via priorityQueue.Update().
+		if len(events) == 0 {
```

Comment:
Why do we still need to determine the length of events since the `events` for-loop won't be executed if the events is nil?

Reply:
Functionality is the same, I just want to highlight that the events could be empty. Make sense to you?

Reply:
I see, thanks.